### ISSUE TITLE
Error when using generator functions

### ISSUE THREAD

**Issue by zhaoyao91** [jaredpalmer/tsdx#547]
### Current Behavior

after add some code with generator, it generates some code which lead to 

```
Uncaught ReferenceError: regeneratorRuntime is not defined
```

I tried to solve it with this 

```
https://github.com/microsoft/TypeScript-Babel-Starter/issues/26#issuecomment-437652709
```

but it gave me other error:

```
(babel plugin) Error: Runtime helpers are not enabled. Either exclude the transform-runtime Babel plugin or pass the `runtimeHelpers: true` option. See https://github.com/rollup/rollup-plugin-babel#configuring-babel for more information
```

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| TSDX           | 0.12.3
| TypeScript       | 3.8.2
| Browser          | chrome
| npm/Yarn         | yarn 1.21.1
| Node             | 12.14.1
| Operating System | mac

**Comment by agilgur5** [comment]
TSDX doesn't currently use regenerator. (it supports async via `async-to-promises` which is apparently buggy, so might change that).

So if you need generators, you should install and add `@babel/plugin-transform-regenerator` to your `babelrc` instead of just the transform.

Let me know if that works.

**Comment by zhaoyao91** [comment]
@agilgur5 I've tried this, no luck.
currently I use the facebook's regenerator-runtime as a workaround: https://github.com/facebook/regenerator/tree/master/packages/regenerator-runtime

just add `import "regenerator-runtime/runtime";` to the index.ts file. Not ideal.

**Comment by agilgur5** [comment]
> @agilgur5 I've tried this, no luck.

You tried `@babel/plugin-transform-regenerator`? What error did it give?

You listed here was from `@babel/plugin-transform-runtime`, which is different and does a lot more.

> just add `import "regenerator-runtime/runtime";` to the index.ts file. Not ideal.

Ok, so this works with `@babel/plugin-transform-runtime` right?

Will have to look into why `@babel/plugin-transform-regenerator` doesn't work. Not sure if that's because of the changes deprecating `@babel/polyfill` for `core-js` directly (importing regenerator runtime as you did is correct in this case)

**Comment by devanfarrell** [comment]
I didn't realize this was supposed to be supported. Here is my babel config I've also been using as a work around. https://github.com/devanfarrell/redux-dogma/blob/master/.babelrc

**Comment by zhaoyao91** [comment]
@agilgur5 
after adding `@babel/plugin-transform-runtime`, it still output `Uncaught ReferenceError: regeneratorRuntime is not defined` error.

when using `regenerator-runtime/runtime`, I don't have to add `@babel/plugin-transform-runtime`.

**Comment by zhaoyao91** [comment]
@devanfarrell this may work but it seems not an extensible way since it replace the internal config.

**Comment by agilgur5** [comment]
> after adding `@babel/plugin-transform-runtime`, it still output `Uncaught ReferenceError: regeneratorRuntime is not defined` error.

Ok, you said that already. What about `@babel/plugin-transform-regenerator`?

**Comment by agilgur5** [comment]
So I added an integration test for this in #627 and that made it easier to investigate this.
- TSDX already uses `@babel/plugin-transform-regenerator` internally, it's just only used for generators and async generators, not plain async/await (`async: false`).
- `@babel/plugin-transform-runtime` is also a dependency of TSDX (but not used internally, might be a peerDep of something?) and it's one possible solution here. Configuring `helpers: false` causes your output file to have `import _regeneratorRuntime from '@babel/runtime/regenerator';` added:

```javascript
// .babelrc.js
module.exports = {
  plugins: [
    ['@babel/plugin-transform-runtime', { helpers: false }]
  ]
}
```
I believe this works the same as `runtimeHelpers: true`. I don't think we'll be setting that as a default however, as that's a decision for the author to make (how they want to use helpers/polyfills). So this is a good user-land alternative.

- The other option may be to use `preset-env`'s `useBuiltIns`. I've used it in the past and think that would likely be a better option, but need to do some more testing with it. Will edit with some details.

Closing this issue since there is a way to do it/workaround available.
